task_processing/plugins/mesos/execution_framework.py

```diff
@@ -46,7 +47,7 @@ class TaskMetadata(PRecord):
     agent_id = field(type=str, initial='')
     task_config = field(type=PRecord, mandatory=True)
     task_state = field(type=str, mandatory=True)
-    task_state_ts = field(type=float, mandatory=True)
+    task_state_history = field(type=PMap, mandatory=True)
```

Comment:
for convenience i would add `factory=pmap` - it'll automatically convert python dicts into pmaps on assignment